### ISSUE TITLE
train_layers only works for fc7 and fc8

### ISSUE THREAD

**Issue by jasstionzyf** [kratzert/finetune_alexnet_with_tensorflow#39]
Thanks for your contribution!
when i reuse your code,  and i want to train fc6, fc7,fc8 ,  but the result: 
Cross entropy = nan
if only train fc7,fc8 ,  it works well.
also i have test earlier layers ,, like conv1,conv2..,  it still appear nan
do have have any suggestion to help me fix this?

**Comment by kratzert** [comment]
I just tried to train all layers, or just a few more and everything works. Are you sure you wrote the correct names into the train_layers list? Maybe your problem is that your gradients explode because of an inappropriate learning rate? But it is definitely possible to train all of the layers you want.

Which Python and Tensorflow version are you using?

**Comment by jasstionzyf** [comment]
@kratzert  thanks for your response very much ! 
  i use tensorflow (1.3.0) , python-3.5.3, 
  seems there something wrong at my side, will check deeply again

**Comment by jasstionzyf** [comment]
@kratzert 
i have run your original program against flowers data,  but it still can learn nothing, also  Cross entropy = nan  when i choose fc6,fc7,fc8 to train ,  and also learning_rate i have test 0.2 or 0.05.

**Comment by kratzert** [comment]
0.2 or 0.05??? This is way to high and most probable the cause for your problem. You would I guess never train a network with a learning rate of 0.2 (which translates to 20% of your gradient values will flow into your weights at every step). You usually finetune with lower learning rates, than the model was originally trained on. Try something like 0.001 and again

**Comment by jasstionzyf** [comment]
@kratzert sorry for misunderstanding ,  what i means is that i have both test 0.2 and 0.05,  
also i have test with learning_rate: 0.001,  still get Nan.

maybe the fc6 layer weights not init properly, if not load from npy, the weights seems use default init way ?

**Comment by kratzert** [comment]
Again, it is definitely the problem of your settings. Might be data, learning rate or what ever. I tested the current code repository with Python 3.6 and various tensorflow versions two different datasets (cats-vs-dogs and a private one) and it worked without any problems. 
So I recommend you play around some more with learning rates and what ever (maybe also with your double mean subtraction which could yield to all neurons get only negative activations.)

I'll close this here, since it is no problem of this code actually
